### ISSUE TITLE
LDAP Support for Authentication

### ISSUE THREAD

**Issue by joe-eklund** [LibrePhotos/librephotos#616]
**Describe the enhancement you'd like**
It would be awesome if we could back LibrePhotos with an LDAP server for user authentication.

**Describe why this will benefit the LibrePhotos**
This would allow for LibrePhotos to reuse users from LDAP instead of having it's own internal users auth. I'd love SSO and OIDC support as well, but LDAP is a great place to start I think.

**Comment by carlchan** [comment]
bumping this, would love SSO/OIDC support, or somehow allow general external authentication proxy support to pass a username through to librephotos.  That would be the most flexible I think.

**Comment by esackbauer** [comment]
Same here, LDAP would be great, or any other SSO mechanism to be used with Authelia, Authentik etc.

**Comment by Scot-Survivor** [comment]
Bumping this again, would love to see it.

For those hosting and manging a lot of resources keeping track of them all, and the users can be hard. LDAP helps with this situation.

**Comment by gardiol** [comment]
Agreed. Using proxy auth for many services, would be great to have LP also get it's login stuff from external auth source. SSO / Authelia / Authentik / OAuth / Proxy... anything might do.
